home/service/details.py

```diff
@@ -1,24 +1,54 @@
 from data_platform_catalogue.search_types import MultiSelectFilter, ResultType
 from django.core.exceptions import ObjectDoesNotExist
 
+
 from .base import GenericService
 
 
-class DetailsService(GenericService):
+class DetailsDataProductService(GenericService):
     def __init__(self, urn: str):
         self.urn = urn
         self.client = self._get_catalogue_client()
 
         filter_value = [MultiSelectFilter("urn", [urn])]
-        search_results = self.client.search(
-            query="", page=None, filters=filter_value)
+        search_results = self.client.search(query="", page=None, filters=filter_value)
 
         if not search_results.page_results:
             raise ObjectDoesNotExist(urn)
 
         self.result = search_results.page_results[0]
+        self.data_product_name = self.result.name
```

Comment:
this only gets used once, can you just reference `self.result.name`? or rename `self.result` to `self.data_product_search_result` maybe?